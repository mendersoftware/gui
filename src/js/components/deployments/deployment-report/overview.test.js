// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import React from 'react';

import { render } from '../../../../../tests/setupTests';
import DeploymentOverview from './overview';

describe('DeploymentOverview Component', () => {
  it('renders correctly', async () => {
    const deployment = {
      name: 'test deployment',
      artifact_name: 'test',
      created: '2019-01-01',
      devices: { a: { id: '13' } },
      finished: '2019-01-01',
      group: 'testGroup',
      statistics: { status: {} }
    };
    const { baseElement } = render(<DeploymentOverview devicesById={{}} deployment={deployment} tenantCapabilities={{ hasFullFiltering: true }} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
  });
});
