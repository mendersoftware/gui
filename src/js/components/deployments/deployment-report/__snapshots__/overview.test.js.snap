// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentOverview Component renders correctly 1`] = `
.emotion-0 .key>b {
  background-color: #f7f7f7;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.1607rem;
}

.emotion-1 iconButton {
  margin-right: 8px;
}

.emotion-3 {
  max-width: 100%;
  font-family: Lato,sans-serif;
  font-size: 0.7545rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 24px;
  color: rgba(10, 10, 11, 0.78);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  opacity: 0.5;
  font-size: 0.675rem;
  height: 18px;
}

.emotion-3.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-3 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #bcbcbc;
  font-size: 0.6964rem;
}

.emotion-3 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: rgb(35, 85, 94);
}

.emotion-3 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: rgb(65, 10, 46);
}

.emotion-3 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.5804rem;
}

.emotion-3 .MuiChip-icon {
  margin-left: 4px;
  margin-right: -4px;
  font-size: 18px;
  color: #bcbcbc;
}

.emotion-3 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(10, 10, 11, 0.26);
  font-size: 16px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
  margin-right: 4px;
  margin-left: -4px;
}

.emotion-3 .MuiChip-deleteIcon:hover {
  color: rgba(10, 10, 11, 0.4);
}

.emotion-4 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 8px;
  padding-right: 8px;
  white-space: nowrap;
}

.emotion-6 {
  margin-left: 96px;
}

.emotion-6svg {
  margin-left: 8px;
}

.emotion-8 {
  background-color: #f7f7f7;
}

.emotion-8:after {
  border-right: 20px solid;
  border-right-color: #f7f7f7;
}

<div
  class="report-container margin-top-large margin-bottom-large"
>
  <div
    class="break-all two-columns emotion-0  column-data "
    style="align-items: center; align-self: flex-start; grid-template-columns: minmax(140px, 1fr) minmax(220px, 1fr); max-width: 25vw;"
  >
    <div
      class="align-right  muted"
    >
      <b>
        Release
      </b>
    </div>
    <div
      class="flexbox "
    >
      <a
        class="flexbox centered"
        href="/releases/test"
        rel="noopener noreferrer"
        style="font-weight: 500;"
        target="_blank"
      >
        test
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall margin-left-small emotion-1"
          data-testid="LaunchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
    </div>
    <div
      class="align-right  muted"
    >
      <b>
        Target device(s)
      </b>
    </div>
    <div
      class="flexbox "
    >
      <a
        class="flexbox centered"
        href="/devices?inventory=group:eq:testGroup"
        rel="noopener noreferrer"
        style="font-weight: 500;"
        target="_blank"
      >
        test deployment
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall margin-left-small emotion-1"
          data-testid="LaunchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault margin-left uppercased emotion-3"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall emotion-4"
          >
            static
          </span>
        </div>
      </a>
    </div>
    <div
      class="align-right  muted"
    >
      <b>
        Category
      </b>
    </div>
    <div
      class="flexbox "
    >
      <div
        title="Software update"
      >
        Software update
      </div>
    </div>
  </div>
  <div
    class="flexbox column"
  >
    <div
      class="break-all two-columns emotion-0  column-data "
      style="align-items: center; align-self: flex-start; grid-template-columns: minmax(140px, 1fr) minmax(220px, 1fr); max-width: 25vw;"
    >
      <div
        class="align-right  muted"
      >
        <b>
          Created at
        </b>
      </div>
      <div
        class="flexbox "
      >
        <time
          datetime="2019-01-01T00:00:00+00:00"
        >
          2019-01-01 00:00
        </time>
      </div>
    </div>
    <a
      class="margin-top-small flexbox center-aligned emotion-6"
    >
      Schedule details 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-1"
        data-testid="ArrowDropDownCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4c4.41 0 8 3.59 8 8s-3.59 8-8 8-8-3.59-8-8 3.59-8 8-8m0-2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13-4-4h8z"
        />
      </svg>
    </a>
  </div>
  <div
    class="statusLarge flexbox centered"
  >
    <img
      src="largeFail.png"
    />
    <div
      class="statusWrapper flexbox centered emotion-8"
    >
      <div
        class="statusWrapperMessage"
      >
        <b
          class="red"
        >
          0
        </b>
         
        devices
         updated successfully
      </div>
    </div>
  </div>
</div>
`;
